cli/azd/internal/repository/app_init.go

```diff
@@ -1081,6 +1109,59 @@ func promptSpringBootVersion(console input.Console, ctx context.Context) (string
 	}
 }
 
+func promptBindingDestination(console input.Console, ctx context.Context, bindingName string) (string, error) {
+	for {
+		destination, err := console.Prompt(ctx, input.ConsoleOptions{
+			Message: fmt.Sprintf("Input the value for spring.cloud.stream.bindings.%s.destination", bindingName),
+			Help:    "Hint: Azure Eventhubs Name, please also ensure application properties is well configured.",
+		})
+		if err != nil {
+			return "", err
+		}
+		if IsValidEventhubsName(destination) {
+			return destination, nil
+		} else {
+			console.Message(ctx, "Invalid destination. Please choose another name.")
+		}
+	}
+}
+
+func IsValidEventhubsName(name string) bool {
+	// up to 256 characters
+	if len(name) == 0 || len(name) > 256 {
+		return false
+	}
+	// contain letters, numbers, periods (.), hyphens (-), and underscores (_)
+	// must begin and end with a letter or number
+	regex := regexp.MustCompile(`^[a-zA-Z0-9][a-zA-Z0-9._-]*[a-zA-Z0-9]$`)
```

Comment:
How about moving it out of func to avoid duplicated calculation?
Here is an example:

> ![image](https://github.com/user-attachments/assets/abcad0a3-f507-45a3-85c3-d9c0662291f0)

Reply:
Updated.